pkgs/development/python-modules/soundcard/default.nix

```diff
@@ -0,0 +1,49 @@
+{
+  buildPythonPackage,
+  cffi,
+  fetchPypi,
+  lib,
+  libpulseaudio,
+  numpy,
+  setuptools,
+  testers,
+}:
+let
+  pname = "SoundCard";
+  version = "0.4.3";
+in
+buildPythonPackage {
+  inherit pname version;
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-QQg1UUuhCAmAPLmIfUJw85K1nq82WRW7lFFq8/ix0Dc=";
+  };
+
+  patchPhase = ''
+    substituteInPlace soundcard/pulseaudio.py \
+      --replace "'pulse'" "'${libpulseaudio}/lib/libpulse.so'"
+  '';
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    cffi
+    numpy
+  ];
+
+  # doesn't work because there are not many soundcards in the
+  # sandbox. See VM-test
+  # pythonImportsCheck = [ "soundcard" ];
+
+  passthru.tests.vm-with-soundcard = testers.runNixOSTest ./test.nix;
+
+  meta = with lib; {
+    description = "A Pure-Python Real-Time Audio Library";
+    homepage = "https://github.com/bastibe/SoundCard";
+    changelog = "https://soundcard.readthedocs.io/en/latest/#changelog";
```

Comment:
It might be better to change it to
`"https://github.com/bastibe/SoundCard/blob/${version}/README.rst#changelog"`
since we can use the current version this way.

Reply:
do you have a suggestion? it looks to me like there are no tags available?